src/lib/components/ui/ScrollView/ScrollView.jsx

```diff
@@ -0,0 +1,156 @@
+  const scrollViewViewportEl = useRef(null);
+
+  const handleScrollViewState = (currentPosition) => {
+    const isScrolledAtStartActive = currentPosition[scrollPositionStart] < 0;
+    const isScrolledAtEndActive = currentPosition[scrollPositionEnd] > 0;
+
+    if (isScrolledAtStartActive !== isScrolledAtStart) {
+      setIsScrolledAtStart(isScrolledAtStartActive);
+    }
+
+    if (isScrolledAtEndActive !== isScrolledAtEnd) {
+      setIsScrolledAtEnd(isScrolledAtEndActive);
+    }
+  };
+
+  useLoadResize(
+    (currentPosition) => (handleScrollViewState(currentPosition)),
+    [isScrolledAtStart, isScrolledAtEnd],
+    scrollViewContentEl,
+    scrollViewViewportEl,
+    debounce,
+  );
+
+  useScrollPosition(
+    (currentPosition) => (handleScrollViewState(currentPosition)),
+    [isScrolledAtStart, isScrolledAtEnd],
+    scrollViewContentEl,
+    scrollViewViewportEl,
+    debounce,
+  );
+
+  const setAlpha = (rgba, alpha) => {
+    const newColor = { ...rgba };
+    newColor.alpha = alpha;
+    return newColor;
+  };
+
+  const rgbaToString = (rgba) => (
+    `rgba(${rgba.red}, ${rgba.green}, ${rgba.blue}, ${rgba.alpha})`
+  );
+
+  const shadowStartColor = rgbaToString(shadowColor);
+  const shadowEndColor = rgbaToString(setAlpha(shadowColor, 0));
+
+  const startShadowCss = customStartShadow
+    ? { ...customStartShadow }
+    : {
+      backgroundImage: `linear-gradient(${direction === 'horizontal' ? 'to right' : 'to bottom'}, ${shadowStartColor}, ${shadowEndColor})`,
+    };
+  const endShadowCss = customEndShadow
+    ? { ...customEndShadow }
+    : {
+      backgroundImage: `linear-gradient(${direction === 'horizontal' ? 'to left' : 'to top'}, ${shadowStartColor}, ${shadowEndColor})`,
+    };
+
+  return (
+    <div
+      className={[
+        styles.root,
+        isScrolledAtStart ? styles.isScrolledAtStart : '',
+        isScrolledAtEnd ? styles.isScrolledAtEnd : '',
+        direction === 'horizontal' ? styles.isRootHorizontal : styles.isRootVertical,
+      ].join(' ')}
+    >
+      <div className={styles.viewport} ref={scrollViewViewportEl}>
+        <div className={styles.content} ref={scrollViewContentEl}>
+          {children}
+        </div>
+      </div>
+      <div
+        className={styles.shadowStart}
+        style={{
+          ...startShadowCss,
+          height: direction === 'horizontal' ? 'auto' : shadowSize,
+          width: direction === 'horizontal' ? shadowSize : 'auto',
+        }}
+        aria-hidden
+      />
+      <div
+        className={styles.shadowEnd}
+        style={{
+          ...endShadowCss,
+          height: direction === 'horizontal' ? 'auto' : shadowSize,
+          width: direction === 'horizontal' ? shadowSize : 'auto',
+        }}
+        aria-hidden
+      />
+    </div>
+  );
+};
+
+ScrollView.defaultProps = {
+  customEndShadow: undefined,
+  customStartShadow: undefined,
+  debounce: 50,
+  direction: 'vertical',
+  shadowColor: {
```

Comment:
the default shadow is not visible enough

Reply:
Are we talking about this?

![image](https://user-images.githubusercontent.com/5614085/82938016-6d3cd900-9f91-11ea-9f59-de00216c93cf.png)

I think it's a good-enough default… We cannot anticipate the nature of content nor the size of its viewport therefore I don't want the scrolling shadows to be too large by default.

At the end of the day, it's always a design decision of particular project how to design the shadows. Compare eg. Apple Music (above, very thin shadows) to TIDAL (below, massive shadows):

![image](https://user-images.githubusercontent.com/5614085/82938433-184d9280-9f92-11ea-9a30-2f235a3fc5b7.png)

![image](https://user-images.githubusercontent.com/5614085/82938392-053ac280-9f92-11ea-8900-11370765a545.png)